org.opent2t.sample.hub.superpopular/com.nest.hub/js/thingTranslator.js

```diff
@@ -195,18 +193,17 @@ class Translator {
      * Puts device details (all fields) payload formatted per nest api
      */
     putDeviceDetailsAsync(deviceType, deviceId, putPayload) {
-        var propertyName = Object.keys(putPayload);
-        var path = this._devicesPath + deviceType + '/' + deviceId + '/' + propertyName[0];
-        return this._firebaseRef.child(path).set(putPayload[propertyName[0]]).then((response) => {
+        var path = this._devicesPath + deviceType + '/' + deviceId;
+        return this._firebaseRef.child(path).set(putPayload).then((response) => {
             if (response === undefined) { //success
                 var result = {
-                    device_id:deviceId
+                    device_id: deviceId
                 };
-                result[propertyName] = putPayload[propertyName[0]];
-                
-                //get temperature scale
-                if (propertyName[0].includes('_temperature_')) {
-                    result['temperature_scale'] = propertyName[0].charAt(propertyName[0].length -1);
+                for (var propertyName in putPayload) {
```

Comment:
for-in looks at non-native properties as well. don't you need a .hasOwnProperty check?

Also Object.keys is generally faster. Not sure about node.js (that's where this code runs, correct)?

Reply:
Fixed